apps/dashboard/app/controllers/scripts_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# The controller for apps pages /dashboard/projects/:project_id/scripts
+class ScriptsController < ApplicationController
+
+  def new
+    @script = Script.new(project_dir: show_script_params[:project_id])
+  end
+
+  def show
+    @script = Script.find(show_script_params[:id], show_script_params[:project_id])
+  end
+
+  # POST  /dashboard/projects/:project_id/scripts
+  def create
+    dir = script_params[:project_id]
+    opts = { project_dir: dir }.merge(create_script_params[:script])
+    @script = Script.new(opts)
+
+    if @script.save
+      redirect_to project_path(params[:project_id]), notice: 'sucess!'
+    else
+      redirect_to project_path(params[:project_id]), alert: @script.errors[:save].last
+    end
+  end
+
+  private
+
+  def create_script_params
+    params.permit({ script: [:title] }, :project_id)
+  end
+
+  def show_script_params
+    params.permit(:id, :project_id)
+  end
```

Comment:
I don't understand the name of the method. Why is it `show_script_params`?

Reply:
To differentiate between `create_script_params`. We accept/anticipate different parameters for accessor methods like `#show` or `#new`. When the controller method is `#create` we expect different parameters. 

Maybe rename to `get_script_params`?

Reply:
`get_script_params` would make more sense to me.